### PR TITLE
Bonding

### PR DESCRIPTION
**Pull request by sqrrm**

### CODE REVIEW

File: src/main/java/bisq/core/dao/bonding/lockup/LockupService.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import static com.google.common.base.Preconditions.checkArgument;
+
+@Slf4j
+public class LockupService {
+    private final WalletsManager walletsManager;
+    private final BsqWalletService bsqWalletService;
+    private final BtcWalletService btcWalletService;
+    private final StateService stateService;
+    @Getter
+    private final MyBlindVoteList myBlindVoteList = new MyBlindVoteList();
```
Comment (ManfredKarrer) [inline src/main/java/bisq/core/dao/bonding/lockup/LockupService.java:61]: I assume that is not needed

File: src/main/java/bisq/core/dao/bonding/lockup/LockupService.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+            outputStream.write(OpReturnType.LOCKUP.getType());
+            outputStream.write(Version.LOCKUP_VERSION);
+            outputStream.write(lockupTime >>> 8);
+            outputStream.write(lockupTime);
+            // TODO: handle short data
+            // Pushdata of <= 4 bytes is converted to int when returned from bitcoind and not handled the way we
+            // require by btcd-cli4j
+            // Write an extra byte to avoid the asm conversion to int in bitcoind
+            outputStream.write(0);
```
Comment (ManfredKarrer) [inline src/main/java/bisq/core/dao/bonding/lockup/LockupService.java:124]: I think we should add a subType anyway to have more flexibility for diff. types of locks

**Review by ManfredKarrer — CHANGES_REQUESTED** [review]

**Comment by ManfredKarrer** [comment]
Looks good all but I need a bit more time to review it with checked out code, probably tomorrow. Is it in sync with latest voting branch code?

**Comment by sqrrm** [comment]
Yes, this is all on top of voting so just checking out core, desktop and common bonding branch should work. Protobuf needs to be regenerated and it will break earlier versions so you'll need to start from fresh genesis if you want to test it.

**Pull request merged**